google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java

```diff
@@ -2532,6 +2639,46 @@ Blob create(
    */
   URL signUrl(BlobInfo blobInfo, long duration, TimeUnit unit, SignUrlOption... options);
 
+  /**
+   * Generates a URL and a map of fields that can be specified in a form to submit a POST request.
+   * The returned map includes a signature which must be provided with the request. Generating a
+   * presigned POST policy requires a service account signer. If an instance of {@link
+   * com.google.auth.ServiceAccountSigner} was passed to {@link StorageOptions}' builder via {@code
+   * setCredentials(Credentials)} or the default credentials are being used and the environment
+   * variable {@code GOOGLE_APPLICATION_CREDENTIALS} is set, generatPresignedPostPolicyV4 will use
+   * that credentials to sign the URL. If the credentials passed to {@link StorageOptions} do not
+   * implement {@link ServiceAccountSigner} (this is the case, for instance, for Google Cloud SDK
+   * credentials) then {@code signUrl} will throw an {@link IllegalStateException} unless an
+   * implementation of {@link ServiceAccountSigner} is passed using the {@link
+   * PostPolicyV4Option#signWith(ServiceAccountSigner)} option.
+   *
+   * <p>Example of generating a presigned post policy which has the condition that only jpeg images
+   * can be uploaded, and applies the public read acl to each image uploaded:
+   *
+   * <pre>{@code
+   * PostFieldsV4 fields = PostFieldsV4.newBuilder().setAcl("public-read").build();
+   * PostConditionsV4 conditions = PostConditionsV4.newBuilder().addContentTypeCondition(ConditionV4Type.MATCHES, "image/jpeg").build();
+   *
+   * PostPolicyV4 policy = storage.generatePresignedPostPolicyV4(
+   *     BlobInfo.newBuilder("my-bucket", "my-object").build(),
+   *     fields, conditions, 7, TimeUnit.DAYS);
+   *
+   * }</pre>
+   *
+   * @param blobInfo the blob uploaded in the form
+   * @param fields the fields specified in the form
+   * @param conditions which conditions every upload must satisfy
+   * @param duration how long until the form expires, in milliseconds
+   * @param options optional post policy options
+   */
+  PostPolicyV4 generatePresignedPostPolicyV4(
+      BlobInfo blobInfo,
+      PostFieldsV4 fields,
```

Comment:
fields and conditions should be optional. I want to say these should be under options.